src/security.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * Created by sam on 27.02.2017.
+ */
+
+
+import {retrieve, store} from './session';
+import {fire} from './event';
+
+export const GLOBAL_EVENT_USER_LOGGED_IN = 'USER_LOGGED_IN';
+export const GLOBAL_EVENT_USER_LOGGED_OUT = 'USER_LOGGED_OUT';
+
+export interface IUser {
+  /**
+   * user name
+   */
+  readonly name: string;
+  /**
+   * list of roles the user is associated with
+   */
+  readonly roles: string[];
+}
+
+export const ANONYMOUS_USER: IUser = {name: 'anonymous', roles: ['anonymous']};
+
+/**
+ * whether the user is logged in
+ * @returns {boolean}
+ */
+export function isLoggedIn() {
+  return retrieve('logged_in') === true;
+}
+
+/**
+ * stores the given user information
+ * @param user
+ */
+export function login(user: {name: string}) {
```

Comment:
Stupid question: Am I logged in, if I import/call this function with any user name? Or even easier, if I add these variables to the store I have access to the full functionality? How/When is the permission checked?

(I just want to make sure that there is no obvious security hole, when we implementing a sophisticated permission system.)